store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreVisibilityTest.java

```diff
@@ -0,0 +1,339 @@
+
+        // Then
+        assertThat(aggregatedResults)
+                .isNotEmpty()
+                .hasSize(1)
+                .first()
+                .matches(e -> e.getProperty(PROPERTY_1).equals(2), "property is aggregated");
+
+        assertThat(concatenatedResults)
+                .isNotEmpty()
+                .hasSize(2)
+                .allMatch(e -> e.getProperty(PROPERTY_1).equals(1), "property value is 1");
+
+        assertThat(noAuthResults).isEmpty();
+    }
+
+    @Test
+    public void shouldGetDataWhenVisibilityPropertyNamesAndAuthsAreDifferent() throws Exception {
+        // Given
+        addGraphs(VISIBILITY_1, VISIBILITY_2);
+        addElements(GRAPH_ID_A, VISIBILITY_1, PUBLIC);
+        addElements(GRAPH_ID_B, VISIBILITY_2, PRIVATE);
+
+        // When
+        Iterable<? extends Element> aggregatedResults = (Iterable<? extends Element>) federatedGraph.execute(
+                new FederatedOperation.Builder()
+                        .op(new GetAllElements())
+                        .build(), USER);
+        Iterable<? extends Element> concatenatedResults = (Iterable<? extends Element>) federatedGraph.execute(
+                new FederatedOperation.Builder()
+                        .op(new GetAllElements())
+                        .mergeFunction(getDefaultMergeFunction())
+                        .build(), USER);
+        Iterable<? extends Element> noAuthResults = (Iterable<? extends Element>) federatedGraph.execute(
+                new FederatedOperation.Builder()
+                        .op(new GetAllElements())
+                        .build(), new User());
+
+        // Then
+        assertThat(aggregatedResults)
+                .isNotEmpty()
+                .hasSize(1)
+                .first()
+                .matches(e -> e.getProperty(PROPERTY_1).equals(1), "property is only visible once");
+
+        assertThat(concatenatedResults)
+                .isNotEmpty()
+                .hasSize(1)
+                .allMatch(e -> e.getProperty(PROPERTY_1).equals(1), "property value is 1");
+
+        assertThat(noAuthResults).isEmpty();
+    }
+
+    @Test
+    public void shouldMergeVisibilityPropertyWhenVisibilityPropertyNamesAreDifferentAndGroupNameIsSame() throws Exception {
+        // Given
+        addGraphs(VISIBILITY_1, VISIBILITY_2);
+        addElements(GRAPH_ID_A, VISIBILITY_1, PUBLIC);
+        addElements(GRAPH_ID_B, VISIBILITY_2, PUBLIC);
+
+        // When
+        Iterable<? extends Element> aggregatedResults = (Iterable<? extends Element>) federatedGraph.execute(
+                new FederatedOperation.Builder()
+                        .op(new GetAllElements())
+                        .build(), USER);
+        Iterable<? extends Element> concatenatedResults = (Iterable<? extends Element>) federatedGraph.execute(
+                new FederatedOperation.Builder()
+                        .op(new GetAllElements())
+                        .mergeFunction(getDefaultMergeFunction())
```

Comment:
I thought the default merge function was not concatenation? Also if the merge function is being set to the default, then what does this do? I would have thought the default merge function value would already be the default merge function.
If what's happening is the actual default is not what is returned by "getDefault", then that seems to be needlessly confusing.

Reply:
The `getDefaultMergeFunction` returns the basic merge function that will be used by an Operation that has not been configured to use any other merge function. This is the `ConcatenateListMergeFunction`.

However, `GetAllElements` has had a merge function set by it in `FederatedStore` which overrides the default. It uses the `ApplyViewToElementsFunction`:
https://github.com/gchq/Gaffer/blob/2b1f8d235c38099b9e269565553896a7e2a64356/store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedStore.java#L158

Reply:
That makes sense, however as this test is intended to use `ConcatenateListMergeFunction`, then wouldn't it be better to use it directly, instead of using a method which happens to return it? That way it would be clear at a glance what the merge function being used was, without having to dig into the codebase.

Reply:
I think @GCHQDev404 advised against this? But I agree it would be more readable

Reply:
Checked with @GCHQDev404 and have changed to `ConcatenateListMergeFunction`